Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/LocalSceneDevelopment/LocalSceneDevelopmentController.cs

```diff
@@ -56,9 +73,19 @@ private async UniTaskVoid ConnectToServerAsync(string localSceneWebsocketServer,
 
                     // Switch to the main thread because `TryReloadSceneAsync` requires that
                     await UniTask.SwitchToMainThread(cancellationToken: ct);
-                    await reloadScene.TryReloadSceneAsync(ct,
-                        wsSceneMessage.MessageCase == WsSceneMessage.MessageOneofCase.UpdateScene ?
-                            wsSceneMessage.UpdateScene.SceneId : wsSceneMessage.UpdateModel.SceneId);
+
+                    try
+                    {
+                        // We need to freeze the character movement until the scene is reloaded
+                        globalWorld.AddOrGet(playerEntity, new StopCharacterMotion());
+
+                        await reloadScene.TryReloadSceneAsync(ct,
```

Comment:
Shall we set a timeout?

Reply:
Now that you improved the awaiting inside `ECSReloadScene` the timeout would no longer be needed, right?

Reply:
I see, makes sense, maybe we can leave a 5 seconds timeout